pkg/keys/ed25519.go

```diff
@@ -43,8 +50,22 @@ func (e *ed25519Verifier) Verify(msg, sig []byte) error {
 	return nil
 }
 
-func (e *ed25519Verifier) MarshalPublicKey() *data.PublicKey {
-	return e.key
+func (e *ed25519Verifier) MarshalPublicKey() (*data.PublicKey, error) {
+	publicKey, err := json.Marshal(ed25519Verifier{
+		PublicKey: data.HexBytes(e.PublicKey),
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	key := &data.PublicKey{
+		Type:       data.KeyTypeEd25519,
+		Scheme:     data.KeySchemeEd25519,
+		Algorithms: data.HashAlgorithms,
+		Value:      publicKey,
+	}
+
+	return key, nil
```

Comment:
I actually don't think there's a lot of value to re-marshalling in this method; I'd revert this change.